pkg/apis/extensions/validation/operatingsystemconfig.go

```diff
@@ -117,6 +119,25 @@ func ValidateContainerdConfig(config *extensionsv1alpha1.ContainerdConfig, purpo
 
 var availableCapabilities = sets.New(extensionsv1alpha1.PullCapability, extensionsv1alpha1.ResolveCapability, extensionsv1alpha1.PushCapability)
 
+var digitsRegex = regexp.MustCompile(`^\d+$`)
+var portRegexp = regexp.MustCompile(`^([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$`)
+
+// validateHostPort check that host and optional port format is `<host>[:<port>]`
+func validateHostPort(hostPort string) []string {
+	var errs []string
+	host := hostPort
+	if index := strings.LastIndexByte(hostPort, ':'); index != -1 {
+		port := hostPort[index+1:]
+		if digitsRegex.MatchString(port) {
```

Comment:
Is there a valid use-case where non-digits follow a colon or why do we need to check this regex first?

Reply:
The idea here is that if hostPort is for example `http://domain.io`, we'd want to log that `http://domain.io` is not a valid subdomain rather than `domain.io` is not a valid port.

But if hostPort is `domain.io:abc`, we'd like to log that `abc` is not a valid port.